src/gui/qgsattributeform.cpp

```diff
@@ -739,7 +742,13 @@ void QgsAttributeForm::onAttributeChanged( const QVariant &value )
 
   updateConstraints( eww );
 
-  emit attributeChanged( eww->field().name(), value );
+  if ( !signalEmitted )
+  {
+    Q_NOWARN_DEPRECATED_PUSH
+    emit attributeChanged( eww->field().name(), value );
+    Q_NOWARN_DEPRECATED_PUSH
+    emit widgetValueChanged( eww->field().name(), value, !mIsSettingFeature );
+  }
```

Comment:
@pblottiere I modify the code to avoid sending the signal twice. In the test (see under), you were expecting the signal to arrive twice (`QCOMPARE( spy2.count(), 2 );`). Was there any reason for this, or does it look fine for you?

Reply:
I assume this is was introduced by accident and the test here just "controls the status quo at time of writing".
@pblottiere if you have a chance to still control before the time of backporting, I'll be happy.

Reply:
@m-kuhn @3nids Sorry guys, I was in vacation these last weeks.

> I assume this is was introduced by accident and the test here just "controls the status quo at time of writing".

It has been along time since I wrote these tests but I think Matthias is right.

So thank you for cleaning this Denis!